src/mc-fle2-insert-update-payload-v2.c

```diff
@@ -322,16 +322,16 @@ mc_FLE2InsertUpdatePayloadV2_decrypt (_mongocrypt_crypto_t *crypto,
    }
 
    _mongocrypt_buffer_resize (
-      &iup->plaintext, fle2aead->get_plaintext_len (ciphertext.len, status));
+      &iup->plaintext, fle2v2->get_plaintext_len (ciphertext.len, status));
    uint32_t bytes_written; /* ignored */
 
-   if (!fle2aead->do_decrypt (crypto,
-                              &iup->userKeyId,
-                              user_key,
-                              &ciphertext,
-                              &iup->plaintext,
-                              &bytes_written,
-                              status)) {
+   if (!fle2v2->do_decrypt (crypto,
+                            &iup->userKeyId,
+                            user_key,
+                            &ciphertext,
+                            &iup->plaintext,
+                            &bytes_written,
+                            status)) {
       return NULL;
    }
    return &iup->plaintext;
```

Comment:
`iup->plaintext.len` needs to be set to `bytes_written` here, since `bytes_written` can now be smaller than the allocated size.

Reply:
Gorram, excellent catch.